analyzers/src/SonarAnalyzer.Common/Rules/ParameterNameMatchesOriginalBase.cs

```diff
@@ -79,5 +80,41 @@ private void VerifyParameters(SyntaxNodeAnalysisContext context, TMethodDeclarat
                 context.ReportDiagnosticWhenActive(Diagnostic.Create(rule, item.actual.GetLocation(), item.actual.ValueText, item.expected.Name, expectedLocation));
             }
         }
+
+        private void VerifyInterfaceMemberParameters(SyntaxNodeAnalysisContext context, IMethodSymbol methodSymbol, TMethodDeclarationSyntax methodSyntax, IMethodSymbol interfaceMember,
+            string expectedLocation)
+        {
+            if (interfaceMember.ContainingType.TypeArguments.IsEmpty)
+            {
+                VerifyParameters(context, methodSyntax, interfaceMember.Parameters, "interface");
+            }
+            else
+            {
+                VerifyGenericInterfaceMemberParameters(context, methodSymbol, methodSyntax, interfaceMember, expectedLocation);
+            }
+        }
+
+        private void VerifyGenericInterfaceMemberParameters(SyntaxNodeAnalysisContext context, IMethodSymbol methodSymbol, TMethodDeclarationSyntax methodSyntax, IMethodSymbol interfaceMember,
+            string expectedLocation)
+        {
+            var parameters = ParameterIdentifiers(methodSyntax).ToList();
+            var interfaceMethodParameters = interfaceMember.Parameters.ToList();
+            var genericInterfaceTypeArguments = interfaceMember.ContainingType.TypeArguments;
+
+            for (var i = 0; i < parameters.Count; i++)
+            {
+                var parameter = parameters[i];
+                var parameterType = methodSymbol.Parameters[i].Type;
+                var interfaceMethodParameter = interfaceMethodParameters[i];
+                if (!parameter.ValueText.Equals(interfaceMethodParameter.Name, Language.NameComparison))
+                {
+                    var matchingGenericInterfaceArg = genericInterfaceTypeArguments.FirstOrDefault(x => parameterType.DerivesOrImplements(x));
+                    if (matchingGenericInterfaceArg == null || matchingGenericInterfaceArg.Name.IndexOf(parameter.ValueText, StringComparison.OrdinalIgnoreCase) != 0)
```

Comment:
Also I would remove any restrictions on the renaming in this case. I would not force the user to use the type name in the parameter name.